integration-cli/docker_cli_prune_unix_test.go

```diff
@@ -111,3 +114,177 @@ func (s *DockerSuite) TestPruneContainerUntil(c *check.C) {
 	c.Assert(strings.TrimSpace(out), checker.Not(checker.Contains), id1)
 	c.Assert(strings.TrimSpace(out), checker.Contains, id2)
 }
+
+func (s *DockerSuite) TestPruneContainerLabel(c *check.C) {
+	out, _ := dockerCmd(c, "run", "-d", "--label", "foo", "busybox")
+	id1 := strings.TrimSpace(out)
+	c.Assert(waitExited(id1, 5*time.Second), checker.IsNil)
+
+	out, _ = dockerCmd(c, "run", "-d", "--label", "bar", "busybox")
+	id2 := strings.TrimSpace(out)
+	c.Assert(waitExited(id2, 5*time.Second), checker.IsNil)
+
+	out, _ = dockerCmd(c, "run", "-d", "busybox")
+	id3 := strings.TrimSpace(out)
+	c.Assert(waitExited(id3, 5*time.Second), checker.IsNil)
+
+	out, _ = dockerCmd(c, "run", "-d", "--label", "foobar", "busybox")
+	id4 := strings.TrimSpace(out)
+	c.Assert(waitExited(id4, 5*time.Second), checker.IsNil)
+
+	// Add a config file of label=foobar, that will have no impact if cli is label!=foobar
+	config := `{"pruneFilters": ["label=foobar"]}`
+	d, err := ioutil.TempDir("", "integration-cli-")
+	c.Assert(err, checker.IsNil)
+	defer os.RemoveAll(d)
+	err = ioutil.WriteFile(filepath.Join(d, "config.json"), []byte(config), 0644)
+	c.Assert(err, checker.IsNil)
+
+	// With config.json only, prune based on label=foobar
+	out, _ = dockerCmd(c, "--config", d, "container", "prune", "--force")
+	c.Assert(strings.TrimSpace(out), checker.Not(checker.Contains), id1)
+	c.Assert(strings.TrimSpace(out), checker.Not(checker.Contains), id2)
+	c.Assert(strings.TrimSpace(out), checker.Not(checker.Contains), id3)
+	c.Assert(strings.TrimSpace(out), checker.Contains, id4)
+
+	out, _ = dockerCmd(c, "container", "prune", "--force", "--filter", "label=foo")
+	c.Assert(strings.TrimSpace(out), checker.Contains, id1)
```

Comment:
`checker.Equals`  on `id1` would be enough in that case (similar comment for the other subcommands)